daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/DecodeV1.scala

```diff
@@ -270,17 +283,26 @@ private[lf] class DecodeV1(minor: LanguageMinorVersion) extends Decode.OfPackage
     private[this] def decodeModuleRef(lfRef: PLF.ModuleRef): (PackageId, ModuleName) = {
       val modName = eitherToParseError(
         ModuleName.fromSegments(lfRef.getModuleName.getSegmentsList.asScala))
-      lfRef.getPackageRef.getSumCase match {
-        case PLF.PackageRef.SumCase.SELF =>
-          (this.packageId, modName)
-        case PLF.PackageRef.SumCase.PACKAGE_ID =>
-          val pkgId = PackageId
-            .fromString(lfRef.getPackageRef.getPackageId)
-            .getOrElse(throw ParseError(s"invalid packageId '${lfRef.getPackageRef.getPackageId}'"))
-          (pkgId, modName)
-        case PLF.PackageRef.SumCase.SUM_NOT_SET =>
+      import PLF.PackageRef.{SumCase => SC}
+      val pkgId = lfRef.getPackageRef.getSumCase match {
+        case SC.SELF =>
+          this.packageId
+        case SC.PACKAGE_ID =>
+          val rawPid = lfRef.getPackageRef.getPackageId
+          PackageId
+            .fromString(rawPid)
+            .getOrElse(throw ParseError(s"invalid packageId '$rawPid'"))
+        case SC.INTERNED_ID =>
+          assertSince(internedIdsVersion, "interned package ID")
+          val iidl = lfRef.getPackageRef.getInternedId
+          def outOfRange = ParseError(s"invalid package ID table index $iidl")
+          val iid = iidl.toInt
```

Comment:
We cannot really represent anything bigger than `In32`, so why do we define it as `Int64` in protobuf message?

Reply:
That's specific to the restrictions of the particular collections library this particular implementation happens to be using; nothing to do with the context of the LF language definition.